Add README for Hosting (#27925)
dotnet/aspnetcore: src/Hosting/README.md

# Hosting

Hosting provides the entry point for configuring and running your application.

> Documentation for Hosting can be found in [ASP.NET Core Web Host](https://docs.microsoft.com/aspnet/core/fundamentals/host/web-host) and [.NET Generic Host in ASP.NET Core](https://docs.microsoft.com/aspnet/core/fundamentals/host/generic-host) docs.

Reviewer: ```suggestion
Documentation for Hosting can be found in the [ASP.NET Core Web Host](https://docs.microsoft.com/aspnet/core/fundamentals/host/web-host) and [.NET Generic Host in ASP.NET Core](https://docs.microsoft.com/aspnet/core/fundamentals/host/generic-host) docs.
```

## Description

The following contains a description of the sub-directories.

`Abstractions`: Contains the main Hosting and Startup interfaces.
`Hosting`: Contains the default implementations for GenericHost, WebHost, and Startup.
`Server.Abstractions`: Contains the interfaces for Server implementations.
`Server.IntegrationTesting`: Contains classes to help deploy servers for testing across the repo.
`TestHost`: Contains a server implementation for in-memory testing against a server.
`WindowsServices`: Contains methods to run an application as a Windows service.

> `samples`: Contains a few sample apps that show examples of using hosting.

Reviewer 2: This doesn't render well. Make it an actual list?

## Development Setup

### Build

> To build this specific project from source, you can follow the instructions [on building a subset of the code](../../BuildFromSource.md#building-a-subset-of-the-code).

Reviewer 2: ```suggestion
To build this specific project from source, you can follow the instructions [on building a subset of the code](../../docs/BuildFromSource.md#building-a-subset-of-the-code).
```

Reviewer 2: This deep link skips all of the pre-reqs. That's going to be confusing.

Author: Both these comments should be raised on the README issue since these sections are copied from the template there.

Reviewer: Yep. Check out https://github.com/dotnet/aspnetcore/pull/27976 where I gave the master doc a spruce up.

### Test

> To run the tests for this project, you can [run the tests on the command line](../../BuildFromSource.md#running-tests-on-command-line) in this directory.

Reviewer 2: Or from VS?

Reviewer: I'd like to avoid these docs being too VS-specific since that is something we want to improve https://github.com/dotnet/aspnetcore/issues/27260.

Reviewer 2: Leaving VS out entirely is excluding a lot of our contributors. We should document what support we have for starters and expand on it later.

Reviewer 2: ```suggestion
To run the tests for this project, you can [run the tests on the command line](../../docs/BuildFromSource.md#running-tests-on-command-line) in this directory.
```

> To run the tests for this project, you can [run the tests on the command line](../../BuildFromSource.md#running-tests-on-command-line) in this directory.

Reviewer 3: Same suggestion as Tratcher plus ~you can~
```suggestion
To run the tests for this project, [run the tests on the command line](../../docs/BuildFromSource.md#running-tests-on-command-line) in this directory.
```
